src/main/software/amazon/event/ruler/ByteMap.java

```diff
@@ -193,7 +191,9 @@ ByteTransition getTransitionForAllBytes() {
      * @return All transitions contained in this map.
      */
     Set<ByteTransition> getTransitions() {
-        return map.values().stream().filter(Objects::nonNull).collect(Collectors.toSet());
+        Set<ByteTransition> result = new HashSet<>(map.values());
+        result.remove(null);
```

Comment:
Huh? Don't know this idiom, why `remove(null)`?

Reply:
basically stream().filter().collect() is creating a set from the values removing all nulls in the process. but there is some overhead going that route (stream is slow in hot paths), so I basically tried to reproduce the same that why I remove null after creating the set, as far as i remember the set can hold one copy of null, I will double check to make sure this is needed.

Reply:
If you end up keeping this, would recommend a short comment or test on why. I can see someone missing this in a refactor fairly easily.

Reply:
I went for having a consistent style on those two, I believe it is easy to understand now.